lib/slather/coverage_file.rb

```diff
@@ -14,10 +14,13 @@ def source_file_pathname
         # TODO: Handle Swift
         path = nil
         if project.source_directory
-          path = Dir["#{project.source_directory}/**/#{base_filename}.m"].first
+          path = Dir["#{project.source_directory}/**/#{base_filename}.{m,mm,cpp,hpp}"].first
           path &&= Pathname(path)
         else
-          pbx_file = project.files.detect { |pbx_file| pbx_file.real_path.basename.to_s == "#{base_filename}.m" }
+          pbx_file = project.files.detect { |pbx_file|
+            t = Regexp.new(Regexp.escape(base_filename.to_s) + "\.(m|mm|cpp|hpp)$")
```

Comment:
Can we put these into an array somewhere, maybe like 
```ruby
def source_file_extensions
  `['m', 'mm', 'cpp', 'hpp']`
end
```

and then use it, ie,

`path = Dir["#{project.source_directory}/**/#{base_filename}.{#{source_file_extensions.join(',')}}"].first`